src/hooks/node/useNodeSettings.ts

```diff
@@ -0,0 +1,28 @@
+import { ConfigOptions } from '@ironfish/sdk'
+import { useEffect, useCallback } from 'react'
+import useAsyncDataWrapper from '../useAsyncDataWrapper'
+
+const useNodeSettings = () => {
+  const [result, promiseWrapper] = useAsyncDataWrapper<Partial<ConfigOptions>>()
+
+  const loadSettings = () =>
+    promiseWrapper(window.IronfishManager.getNodeConfig())
```

Comment:
thought: Related to the question asking if it's convention to make `IronfishManager` methods return promises, but it seems like we're doing extra work here because we're casting the config into a promise when we may not need to.

Unless we need to cast things to promises, it might be cleaner to avoid that in the future.